File: fixtures/scale/lua/large/checks/check_jenkinshash.lua
local jenkinshash = require("ciphers.jenkinshash")
local jenkinshash_spec = require("specs.jenkinshash_spec")
local corpus_hash = require("corpus.corpus_hash")

local function check_jenkinshash()
  local spec = jenkinshash_spec.get()
  for _, text in ipairs(corpus_hash.corpus_hash()) do
    local first = jenkinshash.jenkinshash_digest(text)
    local second = jenkinshash.jenkinshash_digest(text)
    if first ~= second or #first ~= 8 then return false end
  end
  return spec.category == "hash"
end

return { check_jenkinshash = check_jenkinshash }

File: fixtures/scale/typescript/large/src/checks/CheckJenkinshash.ts
import { Jenkinshash } from '../ciphers/Jenkinshash';
import { JenkinshashSpec } from '../specs/JenkinshashSpec';
import { CorpusHash } from '../corpus/CorpusHash';

export class CheckJenkinshash {
  static run(): boolean {
    const spec = JenkinshashSpec.get();
    for (const text of CorpusHash.texts()) {
      const first = Jenkinshash.digest(text);
      const second = Jenkinshash.digest(text);
      if (first !== second || first.length !== 8) return false;
    }
    return spec.category === 'hash';
  }
}
